### ISSUE TITLE
Is there a way for all nodes to share the most commonly used dependencies while independently using less popular dependencies?

### ISSUE THREAD

**Issue by planb788** [Comfy-Org/ComfyUI-Manager#937]
One common issue we often encounter is that when installing a new plugin, there may be conflicts between the dependency versions it specifies and the existing dependencies on the local system.
Is there a way to create a virtual environment where all plugins share the same common dependencies (such as the torch library), and then for each plugin's dependencies that are not in the common dependencies, create an additional separate virtual environment, so that each plugin can enjoy its own independent dependencies plus the shared common dependencies?
This approach maintains the independence of the environment while also reducing redundant dependencies to some extent.

**Comment by ltdrdata** [comment]
While it's possible to separate environments using venv, you can only run one environment at a time, so you can't use two custom nodes that cause conflicts simultaneously.

**Comment by planb788** [comment]
> While it's possible to separate environments using venv, you can only run one environment at a time, so you can't use two custom nodes that cause conflicts simultaneously.

Indeed, that's the case. So currently, if there are conflicting nodes, can we only choose one of the two? It seems like this might be an unsolvable problem.
